systemd-run: update page (#9911)
tldr-pages/tldr: pages/linux/systemd-run.md

# systemd-run

> Run programs in transient scope units, service units, or path-, socket-, or timer-triggered service units.
> More information: <https://www.freedesktop.org/software/systemd/man/systemd-run.html>.

- Run a program as root:

> `systemd-run {{executable}}`

Reviewer: ```suggestion
`systemd-run {{path/to/executable}}`
```
Question to other maintainers: Would this be applicable here, or is the current one fine?

Author: I took a look around, and I feel like commands that resolve `PATH`, for example `which` and `gdb`, use the form I used, while the ones that don't, for example `journalctl` or `ldd`, use your form. `systemd-run` does resolve `PATH`, so I feel like the current form fits better.

Reviewer 2: To make it [convertible](https://github.com/command-line-interface-pages/v2-tooling/tree/main/md-to-clip) to Command Line Interface Page's placeholder it should contain `path/to`.

Author: @EmilySeville7cfg, I have no idea what that means :stuck_out_tongue: So, what should I do?

> - Run a program in the user service, keeping the working (d)irectory and waiting until it exits:

Reviewer: ```suggestion
- Run a program in the user service, keeping the working [d]irectory and waiting until it exits:
```
`[]` is what we use to highlight longer options.

Author: Duh, not sure how did I miss this. All fixed.

`systemd-run --user -d {{executable}}`

> - Provide a custom (u)nit name and pass extra parameters:

Reviewer: ```suggestion
- Provide a custom [u]nit name and pass extra parameters:
```

> `systemd-run -u {{unit_name}} -- {{executable}} {{extra parameters ...}}`

Reviewer: ```suggestion
`systemd-run -u {{unit_name}} -- {{executable}} {{extra_parameters ...}}`
```
I am not 100% sure about this, would this be better?

Author: I used the current version because `CONTRIBUTING.md` has `{{path/to/directory1 path/to/directory2 ...}}` as an example, and there can be multiple parameters. What do you think about this?

Reviewer 2: > I am not 100% sure about this, would this be better?

According to yours style guide it's incorrect, ellipsis always contain two items, and not one before `...`.

Reviewer 3: I prefer to `{{argument1 argument2 ...}}`.

Reviewer: > I prefer to `{{argument1 argument2 ...}}`.

Yeah, seems better

> - Share the (t)erminal with the program (allowing interactive input/output) and make sure the execution details (r)emain after the program exits:

Reviewer: ```suggestion
- Share the [t]erminal with the program (allowing interactive input/output) and make sure the execution details [r]emain after the program exits:
```

`systemd-run -rt {{executable}}`

> - Limit the CPU and memory utilization of the process, set other (p)roperties and wait until it exits:

Reviewer: ```suggestion
- Limit the CPU and memory utilization of the process, set other [p]roperties and wait until it exits:
```

> - Limit the CPU and memory utilization of the process, set other [p]roperties and wait until it exits:
> 
> `systemd-run -p MemoryMax={{memory_in_bytes}} -p CPUQuota={{percentage_of_CPU_time}}% -p {{other_property=value}} --wait {{executable}}`

Reviewer 3: It would be better if 
```suggestion
- Set [p]roperties (e.g. CPUQuota, MemoryMax) of the process and wait until it exits:

`systemd-run -p MemoryMax={{memory_in_bytes}} -p CPUQuota={{percentage_of_CPU_time}}% --wait {{executable}}`
```

> - Use the program in a shell (P)ipeline:

Reviewer: ```suggestion
- Use the program in a shell [P]ipeline:
```

> `{{other_executable}} | systemd-run -P {{executable}} | {{other_executable}}`

Reviewer 3: ```suggestion
`{{executable1}} | systemd-run -P {{executable2}} | {{executable3}}`
```
